src/test/scala/org/zalando/spark/jsonschema/SchemaConverterTest.scala

```diff
@@ -110,4 +110,170 @@ class SchemaConverterTest extends FunSuite with Matchers {
+        )
+        val expected = StructType(Array(
+          StructField("array", ArrayType(datatype), nullable = false)
+        ))
+
+        assert(schema === expected)
+    }
+  }
+
+  test("Array of array should be an array of array") {
+
+    val schema = SchemaConverter.convertContent(
+      """
+          {
+            "$$schema": "smallTestSchema",
+            "type": "object",
+            "properties": {
+              "array" : {
+                "type" : "array",
+                "items": {
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          }
+        """
+    )
+    val expected = StructType(Array(
+      StructField("array", ArrayType(ArrayType(StringType)), nullable = false)
+    ))
+
+    assert(schema === expected)
+  }
+
+  test("Array of object should be an array of object") {
+
+    val schema = SchemaConverter.convertContent(
+      """
+          {
+            "$$schema": "smallTestSchema",
+            "type": "object",
+            "properties": {
+              "array" : {
+                "type" : "array",
+                "items": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        """
+    )
+    val expected = StructType(Array(
+      StructField("array", ArrayType(StructType(Seq.empty)), nullable = false)
+    ))
+
+    assert(schema === expected)
+  }
+
+  test("Array of object with properties should be an array of object with these properties") {
+
+    val schema = SchemaConverter.convertContent(
+      """
+          {
+            "$$schema": "smallTestSchema",
+            "type": "object",
+            "properties": {
+              "array" : {
+                "type" : "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "name" : {
+                      "type" : "string"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        """
+    )
+    val expected = StructType(Array(
+      StructField("array", ArrayType(StructType(Seq(StructField("name", StringType, nullable = false )))), nullable = false)
+    ))
+
+    assert(schema === expected)
+  }
+
+  test("Array of unknown type should be an array of object") {
+
+    val schema = SchemaConverter.convertContent(
+      """
+          {
+            "$$schema": "smallTestSchema",
+            "type": "object",
+            "properties": {
+              "array" : {
+                "type" : "array",
+                "items" : {}
+              }
+            }
+          }
+        """
+    )
+    val expected = StructType(Array(
+      StructField("array", ArrayType(StructType(Seq.empty)), nullable = false)
+    ))
+
+    assert(schema === expected)
+  }
+
+  test("Array of various type should be an array of object") {
```

Comment:
Currently we don't support various types for one field, so it shouldn't be supported inside the array as well. Lists in the 'type'-field are only supported to mark a field as nullable, in that case one of the types should be 'null'. So this test should not succeed.